pkg/orchestration/k8s/cluster.go

```diff
@@ -443,3 +443,43 @@ func (c *Cluster) FlattenTopologies(ctx context.Context) (topologies map[string]
 
 	return
 }
+
+// ClosetFullNodeClient returns the closest full node client to the supplied node.
+func (c *Cluster) ClosetFullNodeClient(ctx context.Context, s *bee.Client, r *rand.Rand) (*bee.Client, error) {
```

Comment:
Should be `Closest`, it is a typo

Reply:
Is it possible to have `ClosestFullNode` method on a bee.Client?

Reply:
I agree, it would be logical that bee client provides closest peer address and here to return the client.

Reply:
The client cannot return another node/client. It also cannot tell if a peer is a full node or not (without an extra API call)
It can return the address of the closest peer but this peer might not be a full node. So this is not reliable.
IMO, `cloestFullNode` should be in the orchestration package.

Reply:
If /topology would include the full node status for every connected peer something like bins.bin_xx.connectedPeers[].fullNode, that would be quite handy.

The bee client should return only the address of the closes peer, and the client for that peer can be created wherever is needed.

In any case it is not a crucial thing, it would just make a better separation of concerns.

Reply:
To achieve this without making any API changes, `client.ClosestPeer` can take in a `bin` and will return the closest peer in that bin. Then the caller can iteratively find the closest full node starting at bin 0.
Would this make more sense?

Reply:
That is a very nice idea, like an iterator over peers by proximity. Sound good to me. I would be interested to hear @gacevicljubisa as well, as he has found this improvement.

Reply:
My suggestion was just because separation of concerns, implmentation is yours and it is fine by me. :)

I guess that we could extract bee interface to specific package and then be able to achive that Client can return another one. But this would require refactoring. This we can do in the future. 

Also, just to mention, we have one similar function [ClosestNodeFromMap](https://github.com/ethersphere/beekeeper/blob/a2803bfd744959195139e1f5fea5726bb663322b/pkg/bee/chunk.go#L112).